Vue mount type definitions error on SFCs with only one `define` API used

### Current behavior

Upgrading to Cypress v11 (any minor version) has resulted in TypeScript errors appearing in my component test files, specifically when using mount with an SFC that has only one of the `defineProps`, `defineEmits` and `defineExpose` API's used in its `setup`.

```vue
<script lang="ts" setup>
defineProps<{
  label: string;
}>();
</script>
  
<template>
  <button type="button" :title="label" :aria-label="label">
    <slot />
  </button>
</template>
```

```typescript
import HelloWorld from './HelloWorld.vue'

describe('<HelloWorld />', () => {
  it('renders', () => {
    cy.mount(HelloWorld, {}) // TypeScript Error ts(2769)
  })
})
```

```
$ vue-tsc --noEmit
src/components/HelloWorld.cy.ts:6:14 - error TS2769: No overload matches this call.
  The last overload gave the following error.
    Argument of type 'ComponentPublicInstanceConstructor<{ $: ComponentInternalInstance; $data: {}; $props: Partial<{}> & Omit<Readonly<ExtractPropTypes<__VLS_TypePropsToRuntimeProps<{ label: string; }>>> & VNodeProps & AllowedComponentProps & ComponentCustomProps, never>; ... 10 more ...; $watch<T extends string | ((...args: any) => any...' is not assignable to parameter of type 'ComponentOptionsWithObjectProps<__VLS_TypePropsToRuntimeProps<{ label: string; }>, {}, {}, {}, {}, ComponentOptionsMixin, ComponentOptionsMixin, ... 5 more ..., {}>'.
      Type 'ComponentPublicInstanceConstructor<{ $: ComponentInternalInstance; $data: {}; $props: Partial<{}> & Omit<Readonly<ExtractPropTypes<__VLS_TypePropsToRuntimeProps<{ label: string; }>>> & VNodeProps & AllowedComponentProps & ComponentCustomProps, never>; ... 10 more ...; $watch<T extends string | ((...args: any) => any...' is not assignable to type 'ComponentOptionsBase<Readonly<ExtractPropTypes<__VLS_TypePropsToRuntimeProps<{ label: string; }>>> & { [x: `on${Capitalize<string>}`]: ((...args: any[]) => any) | undefined; }, ... 10 more ..., string>'.
        Types of property 'setup' are incompatible.
          Type '((this: void, props: Readonly<LooseRequired<Readonly<ExtractPropTypes<__VLS_TypePropsToRuntimeProps<{ label: string; }>>>>>, ctx: SetupContext<...>) => void | ... 2 more ... | Promise<...>) | undefined' is not assignable to type '((this: void, props: Readonly<LooseRequired<Readonly<ExtractPropTypes<__VLS_TypePropsToRuntimeProps<{ label: string; }>>> & { [x: `on${Capitalize<string>}`]: ((...args: any[]) => any) | undefined; }>>, ctx: SetupContext<...>) => void | ... 2 more ... | Promise<...>) | undefined'.
            Type '(this: void, props: Readonly<LooseRequired<Readonly<ExtractPropTypes<__VLS_TypePropsToRuntimeProps<{ label: string; }>>>>>, ctx: SetupContext<...>) => void | ... 2 more ... | Promise<...>' is not assignable to type '(this: void, props: Readonly<LooseRequired<Readonly<ExtractPropTypes<__VLS_TypePropsToRuntimeProps<{ label: string; }>>> & { [x: `on${Capitalize<string>}`]: ((...args: any[]) => any) | undefined; }>>, ctx: SetupContext<...>) => void | ... 2 more ... | Promise<...>'.
              Types of parameters 'ctx' and 'ctx' are incompatible.
                Type 'SetupContext<string[]>' is not assignable to type 'SetupContext<{}>'.
                  Type '{}' is missing the following properties from type 'string[]': length, pop, push, concat, and 29 more.

6     cy.mount(HelloWorld, {})
               ~~~~~~~~~~

  node_modules/cypress/vue/dist/index.d.ts:1339:18
    1339 declare function mount<PropsOptions extends Readonly<ComponentPropsOptions>, RawBindings, D extends {}, C extends ComputedOptions = {}, M extends Record<string, Function> = {}, E extends EmitsOptions = Record<string, any>, Mixin extends ComponentOptionsMixin = ComponentOptionsMixin, Extends extends ComponentOptionsMixin = ComponentOptionsMixin, EE extends string = string>(componentOptions: ComponentOptionsWithObjectProps<PropsOptions, RawBindings, D, C, M, E, Mixin, Extends, EE>, options?: MountingOptions<ExtractPropTypes<PropsOptions> & PublicProps, D>): Cypress.Chainable<{
                          ~~~~~
    The last overload is declared here.


Found 1 error in src/components/HelloWorld.cy.ts:6
```

Changing the component definition code to the follow will make the error disappear:

```vue
<script lang="ts" setup>
defineProps<{
  label: string;
}>();
defineEmits<{
  (e: "hello"): void;
}>();
</script>
  
<template>
  <button type="button" :title="label" :aria-label="label">
    <slot />
  </button>
</template>
```

### Desired behavior

TypeScript should not throw an error when mounting a Vue SFC component with the following syntax, regardless of the component definition:

```typescript
import HelloWorld from './HelloWorld.vue'

describe('<HelloWorld />', () => {
  it('renders', () => {
    cy.mount(HelloWorld, {}) // Should not throw a TypeScript error
  })
})
```

### Test code to reproduce

Due to the similarities to #23653 that I submitted a while back, I forked the repo that @lmiller1990 graciously provided there. Turns out, all I had to do was upgrade the package versions to reproduce the error :)

https://github.com/JoostKersjes/cypress-issue-23653/tree/v11-upgrade-type-issue

### Cypress Version

11.2.0 (reproducable for any version >= 11.0.0)

### Node version

v16.8.0

### Operating System

Kubuntu 20.04

### Debug Logs

_No response_

### Other

_No response_

--- a/npm/vue/src/index.ts
+++ b/npm/vue/src/index.ts
@@ -92,6 +92,7 @@ const cleanup = () => {
  * `MountOptions` are modifying, including some Cypress specific options like `styles`.
  * The return type is different. Instead of VueWrapper, it's Cypress.Chainable<VueWrapper<...>>.
  */
+
 type PublicProps = VNodeProps & AllowedComponentProps & ComponentCustomProps
 
 type ComponentMountingOptions<T> = T extends DefineComponent<
@@ -121,7 +122,7 @@ export function mount<V extends {}>(
   options?: MountingOptions<any> & Record<string, any>
 ): Cypress.Chainable<{
   wrapper: VueWrapper<ComponentPublicInstance<V>>
-  component: VueWrapper<ComponentPublicInstance<V>>['vm']
+  vm: VueWrapper<ComponentPublicInstance<V>>['vm']
 }>
 
 // Class component (without vue-class-component) - props
@@ -134,7 +135,7 @@ export function mount<V extends {}, P>(
   options?: MountingOptions<P & PublicProps> & Record<string, any>
 ): Cypress.Chainable<{
   wrapper: VueWrapper<ComponentPublicInstance<V>>
-  component: VueWrapper<ComponentPublicInstance<V>>['vm']
+  vm: VueWrapper<ComponentPublicInstance<V>>['vm']
 }>
 
 // Class component - no props
@@ -146,7 +147,7 @@ export function mount<V extends {}>(
   options?: MountingOptions<any> & Record<string, any>
 ): Cypress.Chainable<{
   wrapper: VueWrapper<ComponentPublicInstance<V>>
-  component: VueWrapper<ComponentPublicInstance<V>>['vm']
+  vm: VueWrapper<ComponentPublicInstance<V>>['vm']
 }>
 
 // Class component - props
@@ -159,7 +160,7 @@ export function mount<V extends {}, P>(
   options?: MountingOptions<P & PublicProps> & Record<string, any>
 ): Cypress.Chainable<{
   wrapper: VueWrapper<ComponentPublicInstance<V>>
-  component: VueWrapper<ComponentPublicInstance<V>>['vm']
+  vm: VueWrapper<ComponentPublicInstance<V>>['vm']
 }>
 
 // Functional component with emits
@@ -168,7 +169,7 @@ export function mount<Props extends {}, E extends EmitsOptions = {}>(
   options?: MountingOptions<Props & PublicProps> & Record<string, any>
 ): Cypress.Chainable<{
   wrapper: VueWrapper<ComponentPublicInstance<Props>>
-  component: VueWrapper<ComponentPublicInstance<Props>>['vm']
+  vm: VueWrapper<ComponentPublicInstance<Props>>['vm']
 }>
 
 // Component declared with defineComponent
@@ -206,51 +207,52 @@ export function mount<
   > &
   Record<string, any>
 ): Cypress.Chainable<{
+
   wrapper: VueWrapper<
-    InstanceType<
-      DefineComponent<
-        PropsOrPropOptions,
-        RawBindings,
-        D,
-        C,
-        M,
-        Mixin,
-        Extends,
-        E,
-        EE,
-        PP,
-        Props,
-        Defaults
-      >
+  InstanceType<
+    DefineComponent<
+      PropsOrPropOptions,
+      RawBindings,
+      D,
+      C,
+      M,
+      Mixin,
+      Extends,
+      E,
+      EE,
+      PP,
+      Props,
+      Defaults
     >
   >
-  component: VueWrapper<
-    InstanceType<
-      DefineComponent<
-        PropsOrPropOptions,
-        RawBindings,
-        D,
-        C,
-        M,
-        Mixin,
-        Extends,
-        E,
-        EE,
-        PP,
-        Props,
-        Defaults
-      >
-    >
-  >['vm']}
 >
+  vm: VueWrapper<
+  InstanceType<
+    DefineComponent<
+      PropsOrPropOptions,
+      RawBindings,
+      D,
+      C,
+      M,
+      Mixin,
+      Extends,
+      E,
+      EE,
+      PP,
+      Props,
+      Defaults
+    >
+  >
+>['vm']
+}>
 
 // component declared by vue-tsc ScriptSetup
-export function mount<T extends DefineComponent<any, any, any, any>>(
+export function mount<T extends DefineComponent<any, any, any, any, any>>(
   component: T,
   options?: ComponentMountingOptions<T>
 ): Cypress.Chainable<{
   wrapper: VueWrapper<InstanceType<T>>
-  component: VueWrapper<InstanceType<T>>['vm']
+  vm: VueWrapper<InstanceType<T>>['vm']
 }>
 
 // Component declared with no props
@@ -278,9 +280,17 @@ export function mount<
   >,
   options?: MountingOptions<Props & PublicProps, D>
 ): Cypress.Chainable<{
-  wrapper: VueWrapper<ComponentPublicInstance<Props, RawBindings, D, C, M, E, VNodeProps & Props>>
-  component: VueWrapper<ComponentPublicInstance<Props, RawBindings, D, C, M, E, VNodeProps & Props>>['vm']
-}> & Record<string, any>
+  wrapper: VueWrapper<
+
+  ComponentPublicInstance<Props, RawBindings, D, C, M, E, VNodeProps & Props>
+> &
+  Record<string, any>
+  vm: VueWrapper<
+
+  ComponentPublicInstance<Props, RawBindings, D, C, M, E, VNodeProps & Props>
+> &
+  Record<string, any>['vm']
+}>
 
 // Component declared with { props: [] }
 export function mount<
@@ -312,7 +322,7 @@ export function mount<
   options?: MountingOptions<Props & PublicProps, D>
 ): Cypress.Chainable<{
   wrapper: VueWrapper<ComponentPublicInstance<Props, RawBindings, D, C, M, E>>
-  component: VueWrapper<ComponentPublicInstance<Props, RawBindings, D, C, M, E>>['vm']
+  vm: VueWrapper<ComponentPublicInstance<Props, RawBindings, D, C, M, E>>['vm']
 }>
 
 // Component declared with { props: { ... } }
@@ -343,27 +353,27 @@ export function mount<
   options?: MountingOptions<ExtractPropTypes<PropsOptions> & PublicProps, D>
 ): Cypress.Chainable<{
   wrapper: VueWrapper<
-    ComponentPublicInstance<
-      ExtractPropTypes<PropsOptions>,
-      RawBindings,
-      D,
-      C,
-      M,
-      E,
-      VNodeProps & ExtractPropTypes<PropsOptions>
-    >
+  ComponentPublicInstance<
+    ExtractPropTypes<PropsOptions>,
+    RawBindings,
+    D,
+    C,
+    M,
+    E,
+    VNodeProps & ExtractPropTypes<PropsOptions>
   >
-  component: VueWrapper<
-    ComponentPublicInstance<
-      ExtractPropTypes<PropsOptions>,
-      RawBindings,
-      D,
-      C,
-      M,
-      E,
-      VNodeProps & ExtractPropTypes<PropsOptions>
-    >
-  >['vm']
+>
+  vm: VueWrapper<
+  ComponentPublicInstance<
+    ExtractPropTypes<PropsOptions>,
+    RawBindings,
+    D,
+    C,
+    M,
+    E,
+    VNodeProps & ExtractPropTypes<PropsOptions>
+  >
+>['vm']
 }>
 
 /**

--- a/npm/vue/src/index.ts
+++ b/npm/vue/src/index.ts
@@ -122,7 +122,7 @@ export function mount<V extends {}>(
   options?: MountingOptions<any> & Record<string, any>
 ): Cypress.Chainable<{
   wrapper: VueWrapper<ComponentPublicInstance<V>>
-  vm: VueWrapper<ComponentPublicInstance<V>>['vm']
+  component: VueWrapper<ComponentPublicInstance<V>>['vm']
 }>
 
 // Class component (without vue-class-component) - props
@@ -135,7 +135,7 @@ export function mount<V extends {}, P>(
   options?: MountingOptions<P & PublicProps> & Record<string, any>
 ): Cypress.Chainable<{
   wrapper: VueWrapper<ComponentPublicInstance<V>>
-  vm: VueWrapper<ComponentPublicInstance<V>>['vm']
+  component: VueWrapper<ComponentPublicInstance<V>>['vm']
 }>
 
 // Class component - no props
@@ -147,7 +147,7 @@ export function mount<V extends {}>(
   options?: MountingOptions<any> & Record<string, any>
 ): Cypress.Chainable<{
   wrapper: VueWrapper<ComponentPublicInstance<V>>
-  vm: VueWrapper<ComponentPublicInstance<V>>['vm']
+  component: VueWrapper<ComponentPublicInstance<V>>['vm']
 }>
 
 // Class component - props
@@ -160,7 +160,7 @@ export function mount<V extends {}, P>(
   options?: MountingOptions<P & PublicProps> & Record<string, any>
 ): Cypress.Chainable<{
   wrapper: VueWrapper<ComponentPublicInstance<V>>
-  vm: VueWrapper<ComponentPublicInstance<V>>['vm']
+  component: VueWrapper<ComponentPublicInstance<V>>['vm']
 }>
 
 // Functional component with emits
@@ -169,7 +169,7 @@ export function mount<Props extends {}, E extends EmitsOptions = {}>(
   options?: MountingOptions<Props & PublicProps> & Record<string, any>
 ): Cypress.Chainable<{
   wrapper: VueWrapper<ComponentPublicInstance<Props>>
-  vm: VueWrapper<ComponentPublicInstance<Props>>['vm']
+  component: VueWrapper<ComponentPublicInstance<Props>>['vm']
 }>
 
 // Component declared with defineComponent
@@ -226,7 +226,7 @@ export function mount<
     >
   >
 >
-  vm: VueWrapper<
+  component: VueWrapper<
   InstanceType<
     DefineComponent<
       PropsOrPropOptions,
@@ -252,7 +252,7 @@ export function mount<T extends DefineComponent<any, any, any, any, any>>(
   options?: ComponentMountingOptions<T>
 ): Cypress.Chainable<{
   wrapper: VueWrapper<InstanceType<T>>
-  vm: VueWrapper<InstanceType<T>>['vm']
+  component: VueWrapper<InstanceType<T>>['vm']
 }>
 
 // Component declared with no props
@@ -285,7 +285,7 @@ export function mount<
   ComponentPublicInstance<Props, RawBindings, D, C, M, E, VNodeProps & Props>
 > &
   Record<string, any>
-  vm: VueWrapper<
+  component: VueWrapper<
 
   ComponentPublicInstance<Props, RawBindings, D, C, M, E, VNodeProps & Props>
 > &
@@ -322,7 +322,7 @@ export function mount<
   options?: MountingOptions<Props & PublicProps, D>
 ): Cypress.Chainable<{
   wrapper: VueWrapper<ComponentPublicInstance<Props, RawBindings, D, C, M, E>>
-  vm: VueWrapper<ComponentPublicInstance<Props, RawBindings, D, C, M, E>>['vm']
+  component: VueWrapper<ComponentPublicInstance<Props, RawBindings, D, C, M, E>>['vm']
 }>
 
 // Component declared with { props: { ... } }
@@ -363,7 +363,7 @@ export function mount<
     VNodeProps & ExtractPropTypes<PropsOptions>
   >
 >
-  vm: VueWrapper<
+  component: VueWrapper<
   ComponentPublicInstance<
     ExtractPropTypes<PropsOptions>,
     RawBindings,

